src/main/java/eu/dissco/core/digitalspecimenprocessor/service/MidsService.java

```diff
@@ -54,88 +57,175 @@ private boolean compliesToMidsTwo(DigitalSpecimen attributes) {
     }
   }
 
+  private boolean compliesToMidsThree(DigitalSpecimen attributes) {
```

Comment:
you could simplify a bit of work here by not saving the values and failing fast instead:
```
  private boolean compliesToMidsThree(DigitalSpecimen attributes) {
    if (!hasValue(attributes.getDwcInstitutionId()) && !taxValuesComply(attributes) && !geographicalValuesComply(attributes)) {
      return false;
    }
    if (PALEO_GEO_DISCIPLINES.contains(attributes.getOdsTopicDiscipline())) {
      return true;
    }
    return isValid(getDwcRecordedById());
  }
```
if hasValue() fails, you don't check taxCalues or geoValues, etc.

Reply:
Refactored and put the checks in a different function